roles/kubernetes/kubeadm/tasks/main.yml

```diff
@@ -92,21 +92,6 @@
     - kubeadm_discovery_address != kube_apiserver_endpoint
   notify: restart kubelet
 
-- name: Update server field in kube-proxy kubeconfig
-  shell: >-
-    {{ bin_dir }}/kubectl --kubeconfig {{ kube_config_dir }}/admin.conf get configmap kube-proxy -n kube-system -o yaml
-    | sed 's#server:.*#server:\ {{ kube_apiserver_endpoint }}#g'
-    | {{ bin_dir }}/kubectl --kubeconfig {{ kube_config_dir }}/admin.conf replace -f -
-  delegate_to: "{{groups['kube-master']|first}}"
-  run_once: true
-  when:
-    - kubeadm_config_api_fqdn is not defined
-    - is_kube_master
-    - kubeadm_discovery_address != kube_apiserver_endpoint
-    - not kube_proxy_remove
-  tags:
-    - kube-proxy
-
```

Comment:
why delete that?

Reply:
The server field should be populated by using kubeadm configuration files correctly instead of as a hack after deploying kube-proxy.